Automattic/wp-calypso
# Sites Sorting: Move '(Retired) TITLE' sites to the bottom of A-Z sort

See https://github.com/Automattic/wp-calypso/issues/68455#issuecomment-1329616750

## Proposed Changes

Our A-Z sorting algorithm currently puts `(Retired) TITLE` sites at the top of our A-Z list:

![image](https://user-images.githubusercontent.com/36432/204366704-2c71e228-e661-4cb5-bab9-70dd1b844ed1.png)

This pull request fiddles with `sortAlphabetically()` to put `(Retired) TITLE` sites at the very end of an alphabetically sorted list.

I don't love the implementation but it seems better than the alternatives I considered:
* I don't think it makes sense to hide `(Retired) TITLE` sites entirely.
* At the moment, we don't have any structured data to work with. It seems unnecessarily complex to add yet another attribute to the site object.
* `useSitesListSorting()` seemed like the most appropriate place to insert this behavior.
* Starting a site title with a special character (the '`(`') seems like it should place it at the end of an alphabetically sorted list, not the beginning.

## Testing Instructions

1. Add yourself to a retired P2.
2. Enable 'Magic' sorting on the Sites page.
3. Verify the retired P2 appears unexpectedly high on the Sites page list.
4. Check out this pull request.
5. Verify the retired P2 now appears at the very bottom of the Sites page list.
